api/authenticate.ts

```diff
@@ -1,92 +1,95 @@
 import connect from './db';
 import * as errors from './errors';
 import config from './config';
-const crypto = require('crypto');
-const bcrypt = require('bcrypt');
+import crypto from 'crypto';
+import bcrypt from 'bcrypt';
 
-//should be extremely unlikely to generate two keys with the same id; retry when you do, up to this many times
+// should be extremely unlikely to generate two keys with the same id; retry when you do, up to this many times
 const maxKeygenRetries = 20;
 
 // Authentication function: return user id if authorized; otherwise throw an exception
 export function alwaysPermitted(request) {
-  return null;
+    return null;
 }
 
-export async function hash(password: string):Promise<string> {
+export async function hash(password: string): Promise<string> {
     const rounds = config.passwordHashing.rounds;
     const salt = await bcrypt.genSalt(rounds);
-    //hash output includes the salt and rounds
+    // hash output includes the salt and rounds
     const hash = await bcrypt.hash(password, salt);
     return hash;
 }
 
-export async function matchesHash(password: string, hash: string):Promise<boolean> {
+export async function matchesHash(password: string, hash: string): Promise<boolean> {
     return bcrypt.compare(password, hash);
 }
 
-export async function randomSessionId(db):Promise<string> {
+export async function randomSessionId(db): Promise<string> {
     for (let i = 0; i < maxKeygenRetries; i++) {
-        //need a cryptographically secure random number, not Math.random()
+        // need a cryptographically secure random number, not Math.random()
```

Comment:
It's best to do code style changes like these all in one separate PR so it's easier to see what you've actually changed.